### PR TITLE
[Deployment Scripts] Add 'Tail' to get-logs path; update error model 

### PR DESCRIPTION
**Pull request by filizt**
This Pr has the updates mentioned in the title. It also updates the generate.ps to generate the .NET SDK with Deployment Scripts tag.

There is one issue that come when generating the Deployment Scripts SDK by running the generate.ps. The issue is that there are changes in DeploymentScriptsOperations which doesn't happen when creating the SDK manually from command line through autorest. I marked these changes in the PR. Please let me know if these are expected.

For reference the command I run manually is: autorest readme.md --csharp --tag=package-deploymentscripts-2019-10-preview --reflect-api-versions --csharp-sdks-folder=C:\Workspace\SDK --verbose

### CODE REVIEW

File: sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/Models/ScriptType.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// regenerated.
+// </auto-generated>
+
+namespace Microsoft.Azure.Management.ResourceManager.Models
+{
+
+    /// <summary>
+    /// Defines values for ScriptType.
+    /// </summary>
+    public static class ScriptType
```
Comment (filizt) [inline sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/Models/ScriptType.cs]: Change happened when using generate.ps. ScriptType class doesn't get generated when I run the SDK generation autorest command manually.

**Review by isra-fel — COMMENTED** [review]
Currently the recommend way to generate is still using `generate.ps1`, but changing `-AutoRestVersion "latest"` to `"2.0.4413"`. Could you try that to see if the result meets you expectation? Thanks

**Comment by isra-fel** [comment]
Hello @filizt . Ping on this

**Comment by filizt** [comment]
When I generate the SDK with AutoRestVersion set as "2.0.4413" I don't see the extra changes I see using the latest autorest version. Then my question is what's the expected version to be used here? All the others (Policy, Subscription, Resources) are using the latest autorest version as I see in the generate.ps1 and their autorest code generation seems to happen without an issue.

---
In reply to: [599889347](https://github.com/Azure/azure-sdk-for-net/pull/10545#issuecomment-599889347) [](ancestors = 599889347)

File: sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/Models/AzurePowerShellScript.cs [hunk 2]
```diff
@@ -206,13 +206,6 @@ public override void Validate()
             {
                 throw new ValidationException(ValidationRules.CannotBeNull, "AzPowerShellVersion");
             }
-            if (ScriptContent != null)
```
Comment (filizt) [inline sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/Models/AzurePowerShellScript.cs]: This change also only happens when setting -AutoRestVersion to "latest"

File: sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/Models/AzureCliScript.cs [hunk 3]
```diff
@@ -206,13 +206,6 @@ public override void Validate()
             {
                 throw new ValidationException(ValidationRules.CannotBeNull, "AzCliVersion");
             }
-            if (ScriptContent != null)
```
Comment (filizt) [inline sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/Models/AzureCliScript.cs]: Same her. Change only happens when setting -AutoRestVersion to "latest"

File: sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/DeploymentScriptsOperations.cs [hunk 4]
```diff
@@ -172,9 +170,9 @@ public async Task<AzureOperationResponse<DeploymentScript>> UpdateWithHttpMessag
             _url = _url.Replace("{resourceGroupName}", System.Uri.EscapeDataString(resourceGroupName));
             _url = _url.Replace("{scriptName}", System.Uri.EscapeDataString(scriptName));
             List<string> _queryParameters = new List<string>();
-            if (Client.ApiVersion != null)
```
Comment (filizt) [inline sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/DeploymentScriptsOperations.cs]: This change only happens when setting -AutoRestVersion to "latest"

**Comment by isra-fel** [comment]
Hi Filiz, I saw the same as you did - for some modules it seems OK, but for others no. Though the recommendation remains the same, to use fixed 2.0.4413 version to generate.

**Comment by filizt** [comment]
Thanks. Regenerated and committed the SDK with AutoRest version "2.0.4413" as per guidance.

---
In reply to: [602425584](https://github.com/Azure/azure-sdk-for-net/pull/10545#issuecomment-602425584) [](ancestors = 602425584)

**Review by isra-fel — APPROVED** [review]

**Pull request merged**